internal/tsetse/checker.ts

```diff
@@ -47,8 +47,18 @@ export class Checker {
    * `nodeKind` node in `nodeHandlersMap` map. After all rules register their
    * handlers, the source file AST will be traversed.
    */
+  on(nodeKind: ts.SyntaxKind.BinaryExpression,
```

Comment:
why these special overloads? will we need another one for every `ts.SyntaxKind`?

Reply:
I'll leave that up to @bowenni